packages/openapi-generator/src/parser/parameters.ts

```diff
@@ -26,12 +26,25 @@ export function parseParameters(
   parameters = filterDuplicateParams(parameters);
   parameters = reorderParameters(parameters);
   parameters = renameEquallyNamedParams(parameters);
-  return parameters.map(param => ({
+  return parameters.map(param => toOpenApiParameter(param, refs));
+}
+
+function toOpenApiParameter(
+  param: OpenAPIV3.ParameterObject,
+  refs: $Refs
+): OpenApiParameter {
+  const enumValue = resolveObject(param.schema, refs)?.enum;
+  const ret = {
     ...param,
     name: camelCase(param.name),
     // TODO: Check whether types are correct here and whether we can use union types here.
-    type: getType(resolveObject(param.schema, refs)?.type?.toString())
-  }));
+    type: getType(resolveObject(param.schema, refs)?.type?.toString()),
+    enum: resolveObject(param.schema, refs)?.enum
```

Comment:
Is there a good reason to differentiate these here? I couldn't find a case where anywhere further down in the code, where both are relevant at the same time. I think we should parse the enum type here already and transform it to a union type. This would also simplify the code for the wrapper.